dramatiq/broker.py

```diff
@@ -317,7 +317,7 @@ class Consumer:
     Consumers and their MessageProxies are *not* thread-safe.
     """
 
-    def __iter__(self):  # pragma: no cover
+    def __iter__(self) -> "Consumer":  # pragma: no cover
```

Comment:
I think the return type here should be `typing_extensions.Self` because in the subclasses, the return type will be the subclass, not the parent class `Consumer`.

Reply:
Unfortunately, [`Self`](https://docs.python.org/3/library/typing.html#typing.Self) was added with 3.11 but this change wants to type-check on 3.9 and up.

Reply:
I believe if you import it from `typing_extensions` inside of `if TYPE_CHECKING` like [here](https://github.com/Bogdanp/dramatiq/blob/057e03b1727c2813c6e8f5ba209bd3cbec786b0d/dramatiq/actor.py#L41-L42) it should work on 3.9